nlp_profiler/noun_phase_count.py

```diff
@@ -0,0 +1,21 @@
+import en_core_web_sm
+import spacy
+from nlp_profiler.constants import NaN
+nlp = en_core_web_sm.load()
```

Comment:
What is the effect of this line when the module is imported multiple times? Have you tried to run this functionality with a large dataset?

Reply:
I suggest having a look at https://github.com/neomatrix369/nlp_profiler/tree/master/slow-tests/performance_tests, you will see how we have implemented the `line_profiler` and also benchmarked the performance with a test so we know that the performance will be maintained, there might need to be one at the acceptance test level as well not just unit test level.

Let me know if my explanation isn't clear.

Reply:
Can u please share datasets to test on?

Reply:
If you go through all my comments/requests in this PR then you will see what I mean by performance testing the feature/functionality you are working on.

With regards to testing on datasets, in addition to all that I have mentioned, also look at running the notebook: https://github.com/neomatrix369/nlp_profiler/blob/master/notebooks/jupyter/nlp_profiler-large-dataset.ipynb, you may need kaggle authentication for this installed on your local machine.

But it's important you follow the before changes and after changes comparisons to see how much of a time duration was added after your change was introduced. And then also quickly profile the function(s) you wrote. 

I have provided all the hints necessary please ask me if any of my comments are not clear.

Reply:
Performance Test Cases added.

Reply:
I'll take a look and let you know. Tnx